### PR TITLE
Make pipelined TP comm overlap available with mcore

### PR DESCRIPTION
**Pull request by erhoo82**
# What does this PR do ?

Enable the use of pipelined tensor-parallel communication overlap with mcore path.
Removed the condition that enables the use of `ub_tp_comm_overlap` only with `transformer_engine=True`.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by erhoo82** [comment]
jenkins

**Comment by erhoo82** [comment]
jenkins

File: nemo/collections/nlp/models/language_modeling/megatron_base_model.py [hunk 1]
```diff
@@ -792,12 +792,14 @@ def _validate_and_override_config(self):
                 ) % vp_size == 0, 'Make sure the number of model chunks is the same across all pipeline stages.'
 
         if self.cfg.get('ub_tp_comm_overlap', False):
-            if not self.cfg.get('transformer_engine', False) or not self.cfg.get('sequence_parallel', False):
+            if self.cfg.get('ub_tp_comm_overlap', False) and not self.cfg.get('sequence_parallel', False):
```
Comment (athitten) [inline nemo/collections/nlp/models/language_modeling/megatron_base_model.py:795]: @erhoo82 `if self.cfg.get('ub_tp_comm_overlap', False)` seems to be redundant in L795 as it already exists in L794.

**Comment by erhoo82** [comment]
jenkins

**Review by athitten — APPROVED** [review]
LGTM, thanks!

**Pull request merged**